### DISCUSSION TITLE
Duplicating Rows

### DISCUSSION THREAD

**Discussion by kasgithub8** [Budibase/budibase#8223] [Help] [answered]
Hey,

is there a way to save a row with a different Auto ID so that you have the original row and the duplicated one?

Action "Duplicate row" won't work because you can't change the Auto ID in that action.

**Comment by kasgithub8** [comment]
I got it.

**Reply by melohagan** [reply]
Hey @kasgithub8 What was the issue in the end?

**Reply by kasgithub8** [reply]
Well, I'm having a big problem with the Auto ID in a new row.
We got the Idea of adding letters to the Auto IDs of the Contracts for easier display of different versions.

Right now I have 3 columns for that: The Initial Auto ID, NewAutoID and SteadyAutoID

Initial Auto ID: To create the contract
NewAutoID: gets filled by creating the contract to either the same ID (contract) or ID + "a" (template)
SteadyAutoID: also gets filled by creating the contract and is just there so I have that value at all times

My problem is that I can't fill NewAutoID/SteadyAutoID when creating a contract because Auto ID only gets filled after successfully creating the contract. 

Is there any way to work around this?

**Reply by melohagan** [reply]
An automation could do the trick here. **On Row Created** -> **Update Row**

So whenever a row gets created you could have the automation fill in the other custom auto ids.

**Reply by kasgithub8** [reply]
I got that to work, now my next problem is that the relationship column that im tryna save, doesn't work as expected.

`"Client": [
      {
        "_id": "ro_ta_54f6fd44900846b19778481c4ff16f1d_010e52639e0c45c3bcb0ffeebe555d9d",
        "_rev": "5-9938343d8ff5d3429c62a9e4ef629091",
        "Auto ID": 3,
        "Created At": "2022-02-13T06:30:22.156Z",
        "Updated At": "2022-02-13T06:30:22.156Z",
        "Client Company": "A1",
        "tableId": "ta_54f6fd44900846b19778481c4ff16f1d"
      }
    ]`

That's the output of `{{ trigger.row.Client }}`.
`{{ trigger.row.[Client].[0].[Client Company] }}` doesn't work.

**Reply by melohagan** [reply]
If you're using square brackets referencing then don't use the dot.
So try: `{{ trigger.row.Client.0.Client Company }}` OR `{{ trigger.row.Client[0][Client Company] }}`

**Reply by kasgithub8** [reply]
It won't work without square brackets because of the space between Client Company and `{{ trigger.row.Client[0][Client Company] }}` gives the same output of: 
`"Client": [ { "_id": "ro_ta_54f6fd44900846b19778481c4ff16f1d_010e52639e0c45c3bcb0ffeebe555d9d", "_rev": "5-9938343d8ff5d3429c62a9e4ef629091", "Auto ID": 3, "Created At": "2022-02-13T06:30:22.156Z", "Updated At": "2022-02-13T06:30:22.156Z", "Client Company": "A1", "tableId": "ta_54f6fd44900846b19778481c4ff16f1d" } ]`

**Reply by melohagan** [reply]
There is an open issue here: https://github.com/Budibase/budibase/issues/5715 although it's more of a feature of the Handlebars library.

Have you confirmed that it does work if you rename "Client Company" to "ClientCompany" ?

**Reply by kasgithub8** [reply]
> Have you confirmed that it does work if you rename "Client Company" to "ClientCompany" ?

I just did but it does not work.

**Reply by kasgithub8** [reply]
Btw `{{ trigger.row.[Client].[0].[Client Company] }}` works in SMTP Automations

**Reply by melohagan** [reply]
Could you send me some screenshots of your automation and of an example trigger row please?

**Reply by kasgithub8** [reply]
https://user-images.githubusercontent.com/95284278/195539984-3686be74-fc1c-4550-9357-8dbabfabe7c3.png

**Comment by melohagan** [comment, marked as answer]
Thanks @kasgithub8 I found this bug here: https://github.com/Budibase/budibase/issues/7750

If you publish your automation does it work as intended in your published app?

**Reply by kasgithub8** [reply]
I've been trying it the whole time with the published app and just the test automation and both do not work for me.

**Reply by kasgithub8** [reply]
I don't know if it was fixed with the latest but the automation fully works now! Thank you. @melohagan

**Reply by melohagan** [reply]
Interesting, that's great! The joys of keeping software up-to-date 😁

**Reply by kasgithub8** [reply]
@melohagan quick question, why is "Ascending" sorting like this? 
![image](https://user-images.githubusercontent.com/95284278/195828067-ac51c326-9831-4b12-b645-89eefed51c7b.png)

**Reply by melohagan** [reply]
Is the type definitely **Number** and not **string**/**text** ?

**Reply by kasgithub8** [reply]
I had to use Text for this so I could add letters at the end of the numbers.
Is there anything I can do?

**Reply by melohagan** [reply]
There's no way to add a custom sorting algorithm. What you could maybe do is have two columns - one as a number (maybe even the Auto ID?) and another as the string code, that way you could sort on the number but still see the alphanumeric code.
